Sources/Controllers/SpotifyController.swift

```diff
@@ -0,0 +1,202 @@
+    ///   If `config` is `.empty`, the controller will not attempt to connect automatically.
+    ///   The default log level is **`SPTAppRemoteLogLevel.debug`**.
+    ///   - accessTokenDAO: An object conforming to `DAO<String>` that persists the Spotify access token. By default, this optional parameter
+    ///   is nil and the access token is not persisted across app launches.
+    ///   - autoConnect: A Boolean value that determines whether the controller should attempt to connect to Spotify automatically upon initialization.
+    ///   The default value is `true`. If `true` and `config` is not `.empty`, the controller will initiate the connection.
+    ///
+    /// - Warning: Saving the Spotify access token (_or any other sensitive informative)_ using **``UserDefaults``** is not recommended for production apps.
+    ///   Prefer providing a `DAO` that persists values to the Keychain 🔐
+    ///
+    public init(
+        config: SpotifyConfig,
+        accessTokenDAO: (any DAO<String>)? = nil,
+        autoConnect: Bool = true
+    ) {
+        self.config = config
+        self.accessTokenDAO = accessTokenDAO
+        super.init()
+        if autoConnect && config != .empty {
+            connect()
+        }
+    }
+    
+    /// Deep links to the Spotify app where the authorization flow is completed before reopening the host app.
+    ///
+    /// Use `setAccessToken(from url:)` in a SwiftUI `View.onOpenURL(perform:)` modifier to parse and persist the access token.
+    /// This will be called when the Spotify app completes the authorization flow as it uses the `redirectURL` provided in the configuration
+    /// to deep link back to the app that initiated the authorization flow.
+    public func authorize() {
+        // Using an empty string here will attempt to play the user's last song
+        self.remote.authorizeAndPlayURI("")
+    }
+    
+    /// Parses the provided URL to extract and assign the access token, optionally persisting it across app launches.
+    ///
+    /// This method does three things:
+    /// - Parses the provided URL
+    /// - Assigns the access token (if found) to the local `SPTAppRemote.connectionParameters.accessToken`
+    /// - Attempts to persist the access token using the optional `DAO<String>` property.
+    ///
+    /// Use this inside a SwiftUI onOpenURL modifier to parse and save the access token from the URL callback.
+    /// ```
+    /// .onOpenURL { systemController.spotify.setAccessToken(from: $0) }
+    /// ```
+    /// - Important: The access token will **not be persisted** across app launches if an `accessTokenDAO` parameter
+    /// is not provided when initializing `SpotifyController`.
+    public func setAccessToken(from url: URL) {
+        guard
+            let parameters = remote.authorizationParameters(from: url),
+            let newToken = parameters[SPTAppRemoteAccessTokenKey]
+        else {
+            Self.log.error("Failed to parse access token from URL: \(url)")
```

Comment:
I suppose it would be more prudent to implement a privacy level on locations where sensitive data could potentially be logged. 🤓

```suggestion
            Self.log.error("Failed to parse access token from URL: \(url, privacy: .private)")
```